modules/ocf_kvm/manifests/init.pp

```diff
@@ -42,5 +42,14 @@
       ensure => link,
       links  => manage,
       target => '/opt/share/utils/staff/sys/makevm';
+
+    '/opt/share/kvm':
+      ensure => directory;
+
+    '/opt/share/kvm/makevm-ssh-key':
+      content   => file('/opt/puppet/shares/private/makevm-ssh-key'),
```

Comment:
We'd have to somehow get the public key(s) into the netboot image or preseed file though, which would greatly complicate the task.